Source/ExtendedSampler.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+  ==============================================================================
+
+    ExtendedSampler.cpp
+    Created: 27 Jun 2019 4:53:19pm
+    Author:  artyom
+
+  ==============================================================================
+*/
+
+#include "ExtendedSampler.h"
+
+namespace soompler {
+
+ExtendedSound::ExtendedSound (const String& soundName,
+                            AudioFormatReader& source,
+                            const BigInteger& notes,
+                            int midiNoteForNormalPitch,
+                            double attackTimeSecs,
+                            double releaseTimeSecs,
+                            double maxSampleLengthSeconds)
+    : name (soundName),
+      sourceSampleRate (source.sampleRate),
+      midiNotes (notes),
+      midiRootNote (midiNoteForNormalPitch)
+{
+    if (sourceSampleRate > 0 && source.lengthInSamples > 0)
+    {
+        length = jmin ((int) source.lengthInSamples,
+                       (int) (maxSampleLengthSeconds * sourceSampleRate));
+
+        data.reset (new AudioBuffer<float> (jmin (2, (int) source.numChannels), length + 4));
+
+        source.read (data.get(), 0, length + 4, 0, true, true);
+
+        params.attack  = static_cast<float> (attackTimeSecs);
+        params.release = static_cast<float> (releaseTimeSecs);
+    }
+}
+
+ExtendedSound::~ExtendedSound()

```

Comment:
ok, changed it (all fixes for this pr I will actually include in new branch v0.0.3bugf because it is ready to be merged too, dont want to complicate things)